src/Illuminate/Support/Arr.php

```diff
@@ -331,7 +331,17 @@ public static function isAssoc(array $array)
      */
     public static function only($array, $keys)
     {
-        return array_intersect_key($array, array_flip((array) $keys));
+        $attributeData = [];
+
+        foreach ((array) $keys as $key) {
+            if (! $keyValue = data_get($array, $key)) {
```

Comment:
hmmm, this one is to pass the following test:

```
$array = ['name' => 'taylor', 'age' => 26];
$this->assertEquals(['name' => 'taylor'], array_only($array, ['name']));
$this->assertSame([], array_only($array, ['nonExistingKey']));
```

Not sure if there are other cases I missed.